packages/react-core/src/components/TextArea/TextArea.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/FormControls/styles.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /** Additional classes added to the TextArea. */
+  className: PropTypes.string,
+  /** Flag to show if the TextArea is required. */
+  isRequired: PropTypes.bool,
+  /** Flag to show if the TextArea is valid or invalid. */
+  isValid: PropTypes.bool,
+  /** Value of the TextArea. */
+  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  /** A callback for when the TextArea value changes. */
+  onChange: PropTypes.func,
+  /** Custom flag to show that the TextArea requires an associated id or aria-label. */
+  'aria-label': props => {
+    if (!props.id && !props['aria-label']) {
```

Comment:
do you need a prop for id then?

Reply:
I'm not sure if we were going to rely on the consumer to manually provide one or give our own id prop - we didn't do this in the text input component. We should revisit this, for now let's leave as is.

Reply:
You do not need an id prop here. We discussed this with Doug before he left.  We are not requiring the Consumer ti provide an aria label. However, we are warning them if the aria-label or id is missing. The consumer should provide either an aria-label or an id to support accessibility.